core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -0,0 +1,285 @@
+     * In absence of retryAfter, the poller will use pollInterval.
+     *
+     * @param pollInterval Not null  and greater than zero poll interval. It ensure that polling happens only once in given pollInterval.
+     * @param pollOperation to be called by poller. It should not return {@code null}. The response should always have valid {@link com.azure.core.util.polling.PollResponse.OperationStatus}
+     * Mono from pollOperation should never return Mono.error(). If any unexpected
+     * scenario happens in pollOperation, it should catch it and return a valid PollResponse.
+     * This is because poller does not know what to do in case on Mono.error() and user always expect {@link PollResponse}
+     * and we will continue to poll in this scenario.
+     *
+     * @throws NullPointerException If it is {@code null} for {@code  pollInterval pollOperation}.
+     * @throws IllegalArgumentException if {@code  pollInterval} is negative or zero
+     */
+    public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse<T>>> pollOperation) {
+
+        Objects.requireNonNull(pollInterval, "The poll interval input parameter cannot be null.");
+        if (pollInterval.toNanos() <= 0) {
+            throw new IllegalArgumentException("Negative or zero value for poll interval not allowed.");
+        }
+        Objects.requireNonNull(pollOperation, "The poll operation input parameter cannot be null.");
+
+        this.pollInterval = pollInterval;
+        this.pollOperation = pollOperation;
+        this.pollResponse = new PollResponse<>(PollResponse.OperationStatus.NOT_STARTED, null);
+
+        this.fluxHandle = asyncPollRequestWithDelay()
+            .flux()
+            .repeat()
+            .takeUntil(pollResponse -> pollResponse != null && pollResponse.isDone())
+            .share();
+
+        // auto polling start here
+        this.fluxDisposable = fluxHandle.subscribe();
+        this.autoPollingEnabled = true;
+    }
+
+    /**
+     * Create a Poller object with poll interval, poll operation and cancel operation. The polling starts immediately by default and invoke poll operation.
+     *
+     * @param pollInterval Not null poll interval. It ensure that polling happens only once in given pollInterval.
+     * @param pollOperation   to be called by poller. User should never return {@code null}. The response should have valid {@link com.azure.core.util.polling.PollResponse.OperationStatus}
+     * @param cancelOperation cancel operation
+     * @throws NullPointerException If {@code pollerOptions pollOperation} is {@code null}.
+     */
+    public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse<T>>> pollOperation, Consumer<Poller> cancelOperation) {
+        this(pollInterval, pollOperation);
+        this.cancelOperation = cancelOperation;
+    }
+
+    /**
+     * Calls Cancel Operation function if provided.
```

Comment:
```suggestion
     * Attempts to cancel the long-running operation that this Poller represents. This is only possible if the service supports it, 
     * otherwise an {@code UnsupportedOperationException} will be thrown.
```